db/external_sst_file_ingestion_job.cc

```diff
@@ -793,6 +793,18 @@ Status ExternalSstFileIngestionJob::GetIngestedFileInfo(
     file_to_ingest->smallest_internal_key.SetFrom(key);
 
     iter->SeekToLast();
+    if (!iter->Valid()) {
+      // Not Support SeekToLast , have to iter all key
+      // mostly would happen in PlainTable
+      iter->SeekToFirst();
+      auto largest = iter->key();
+      for (; iter->Valid(); iter->Next()) {
+        if (cfd_->internal_comparator().Compare(iter->key(), largest) > 0) {
+          largest = iter->key();
+        }
+      }
+    }
+
```

Comment:
Iter is not Valid() here after the for loop above?

Reply:
You are right. This is a bug , but the iter->key() could still return a correct largest key here, so the code could work.
I will fix it.